### ISSUE TITLE
Set Marta as Default App to Open Folders

### ISSUE THREAD

**Issue by allen-liaoo** [marta-file-manager/marta-issues#861]
Is it possible to make Marta the default app to open folders with? Instead of having finder as the default app.

**Comment by bruno-robert** [comment]
On macOS, finder is very deeply integrated into the system. I haven't seen a single 3rd party file manager offer this option.

I reckon the best way would be to have a kernel extension designed to catch calls to Finder and redirect them to Marta, but this kind of work is long and tedious.

There are some post os stack exchange discussing this very thing:
[How to change the default file manager for OSX?](https://apple.stackexchange.com/questions/73529/how-to-change-the-default-file-manager-for-osx) and [Is there a way to replace Finder altogether?](https://apple.stackexchange.com/questions/19589/is-there-a-way-to-replace-finder-altogether)

**Comment by valrus** [comment]
PathFinder and ForkLift both offer this, though with some limitations — some stuff that points to Finder is baked in at a pretty low level.

PathFinder has it in their preferences so I'm not sure how they do it, but ForkLift offer the following instructions which could maybe be adapted to Marta: https://binarynights.com/manual#fileviewer

**Comment by notDavid** [comment]
Yes, this works for me... set it like this and reboot (logout did not work 100%) :

```
defaults write -g NSFileViewer -string org.yanex.marta
defaults write com.apple.LaunchServices/com.apple.launchservices.secure LSHandlers -array-add '{LSHandlerContentType="public.folder";LSHandlerRoleAll="org.yanex.marta";}'
```

**Comment by JonathanMMann** [comment]
Here it is as a plugin:

plugin {
    id = "com.user.setdefaultfm",
    name = "Set Default File Manager",
    apiVersion = "2.2"
}

local BUNDLE_ID = "sh.marta.Marta"

action {
    id = "enable",
    name = "Set Marta as Default File Manager",
    apply = function()
        martax.execute("/usr/bin/defaults", {"write", "-g", "NSFileViewer", "-string", BUNDLE_ID})
        martax.execute("/usr/bin/defaults", {
            "write",
            "com.apple.LaunchServices/com.apple.launchservices.secure",
            "LSHandlers",
            "-array-add",
            '{"LSHandlerContentType"="public.folder";"LSHandlerRoleAll"="' .. BUNDLE_ID .. '";}'
        })

        martax.alert("Done", "Marta has been set as the default file manager. Please log out and back in for changes to take effect.")
    end
}

action {
    id = "disable",
    name = "Restore Finder as Default",
    apply = function()
        martax.execute("/usr/bin/defaults", {"delete", "-g", "NSFileViewer"})

        martax.alert("Done", "Finder has been restored as the default file manager. Please log out and back in for changes to take effect.")
    end
}

Enjoy!
